flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -245,9 +245,19 @@ public static void deleteJobGraphInKubernetesHA(
     }
 
     public static boolean isHaMetadataAvailable(
-            Configuration conf, KubernetesClient kubernetesClient) {
-        String clusterId = conf.get(KubernetesConfigOptions.CLUSTER_ID);
-        String namespace = conf.get(KubernetesConfigOptions.NAMESPACE);
+            Configuration observeConfig,
+            Configuration deployConfig,
+            KubernetesClient kubernetesClient) {
+
+        if (!FlinkUtils.isKubernetesHAActivated(observeConfig)
```

Comment:
It is a hard time for me to understand why we are verifying the HA enabled again here. AFAIK, we should already done this in the validator.

After more consideration, I find it is useful in `ApplicationReconciler#getAvailableUpgradeMode`. Maybe we should not mix the HA enabled verification and HA metadata existence. It will also make some logging/exception confusing.